src/controller/task.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response, NextFunction } from 'express'
+import TaskModel, { Task } from '../model/task'
+
+export default {
+  async createTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const { title, description, importance, status } = req.body
+      const newTask: Task = await TaskModel.create({
+        title,
+        description,
+        importance,
+        status
+      })
+      res.status(201).send(newTask)
+    } catch (err) {
+      next(err)
+    }
+  },
+
+  async getAllTasks (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const limit = parseInt(<string>req.query.limit)
+      const skip = parseInt(<string>req.query.skip)
+      const taskTotal: number = await TaskModel.countDocuments({})
+      const tasks: Task[] = await TaskModel.find({}).skip(skip).limit(limit).lean()
```

Comment:
Here you could send those requests in parallel by using `Promise.all()`

Reply:
4cde2b3dca863c578f72dbf1fde339d5077cbb4d